app/models/user.rb

```diff
@@ -1,4 +1,29 @@
 class User < ActiveRecord::Base
   has_many :bids
   validates :email, email: true, allow_blank: true
+  before_save :clear_sam_status_if_duns_changed
```

Comment:
Is the fix extracting the logic of `clear_sam_status_if_duns_changed` into a different module/class but retaining the `before_save` callback ... or is there a way to encapsulate without the AR callback?